user/tests/wiring/no_fixture/ble_provisioning.cpp

```diff
@@ -0,0 +1,101 @@
+test(LISTENING_00_DISABLE_LISTENING_MODE) {
+    System.enableFeature(FEATURE_DISABLE_LISTENING_MODE);
+    Network.listen();
+    delay(1000); // Time for system thread to enter listening mode, if any
+    assertFalse(Network.listening());
+}
+
+// Enter listening mode based on the flag
+test(LISTENING_01_ENABLE_LISTENING_MODE) {
+    System.disableFeature(FEATURE_DISABLE_LISTENING_MODE);
+    Network.listen();
+    delay(1000); // Time for system thread to enter listening mode
+    assertTrue(Network.listening());
+    Network.listen(false);
+    delay(1000); // Time for system thread to exit listening mode
+    assertFalse(Network.listening());
+}
+
+// If the flag is enabled while device is in listening mode,
+// device-os should exit listening mode
+test(LISTENING_02_DISABLE_FLAG_WHILE_IN_LISTENING_MODE) {
+    Network.listen();
+    delay(1000); // Time for system thread to enter listening mode
+    assertTrue(Network.listening());
+    System.enableFeature(FEATURE_DISABLE_LISTENING_MODE);
+    delay(1500); // Time for system thread to process the flag
+    assertFalse(Network.listening());
+}
+
+test(LISTENING_03_ENABLE_BLE_PROV_MODE_WHEN_FLAG_SET) {
+    System.enableFeature(FEATURE_DISABLE_LISTENING_MODE);
+    BLE.provisioningMode(true);
+    HAL_Delay_Milliseconds(100);
+    assertTrue(BLE.getProvisioningStatus());
+    assertTrue(BLE.advertising());
+    BLE.provisioningMode(false);
+    HAL_Delay_Milliseconds(100);
+    assertFalse(BLE.getProvisioningStatus());
+    assertFalse(BLE.advertising());
+}
+
+test(LISTENING_04_ENABLE_BLE_PROV_MODE_WHEN_FLAG_CLEARED) {
+    System.disableFeature(FEATURE_DISABLE_LISTENING_MODE);
+    assertEqual(BLE.provisioningMode(true), (int)SYSTEM_ERROR_NOT_ALLOWED);
+    HAL_Delay_Milliseconds(100);
+    assertFalse(BLE.getProvisioningStatus());
+    assertFalse(BLE.advertising());
+    BLE.provisioningMode(false);
+    HAL_Delay_Milliseconds(100);
+    assertFalse(BLE.getProvisioningStatus());
+    assertFalse(BLE.advertising());
+}
+
+test(LISTENING_05_ENABLE_BLE_PROV_AFTER_LISTENING_MODE) {
+    Network.listen();
+    HAL_Delay_Milliseconds(1000); // Time for system thread to enter listening mode
+    assertTrue(Network.listening());
+    SCOPE_GUARD({
+        // Make sure we restore cloud connection after exiting this test because we entered listening mode
+        Particle.connect();
+        waitFor(Particle.connected, HAL_PLATFORM_MAX_CLOUD_CONNECT_TIME);
+        assertTrue(Particle.connected());
+       });
+    System.enableFeature(FEATURE_DISABLE_LISTENING_MODE);
+    HAL_Delay_Milliseconds(1500); // Time for system thread to process the flag
+    assertFalse(Network.listening());
+    BLE.provisioningMode(true);
+    HAL_Delay_Milliseconds(100);
+    assertTrue(BLE.getProvisioningStatus());
+    BLE.provisioningMode(false);
+    HAL_Delay_Milliseconds(100);
+    assertFalse(BLE.getProvisioningStatus());
+    System.disableFeature(FEATURE_DISABLE_LISTENING_MODE);
```

Comment:
Call at the beginning of the test with `SCOPE_GUARD`, so that no matter if the test fails we restore the feature flag.